src/lib.rs

```diff
@@ -587,6 +594,8 @@ impl KafkaJsonToDelta {
                         state.delta_partition_offsets.insert(p, Some(o));
                     }
 
+                    self.try_create_checkpoint(version).await?;
```

Comment:
will we miss a checkpoint if the daemon crashed after the commit but before it was able to write the checkpoint?

Reply:
yes, but these changes are like plan B. E.g. if lambda infra will take longer time, we'll got this as a backup cover :)

Reply:
So there's been not a lot of thoughts into the impl so far

Reply:
i think delayed checkpoint is fine, which is the case for lambda with retry. but we should try to avoid missing an entire checkpoint. If I understand this correctly, once we crashed on producing a checkpoint, there is no mechanism to retry.

given we are not using this in production, i think it's totally fine to leave it as is for now and file a ticket to track it later.